airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/FeatureFlagFetchActivityImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.api.client.generated.WorkspaceApi;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.WorkspaceRead;
+import io.airbyte.featureflag.FeatureFlagClient;
+import io.airbyte.featureflag.FieldSelectionEnabled;
+import io.airbyte.featureflag.Flag;
+import io.airbyte.featureflag.Workspace;
+import jakarta.inject.Singleton;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class FeatureFlagFetchActivityImpl implements FeatureFlagFetchActivity {
+
+  private final WorkspaceApi workspaceApi;
+  private final FeatureFlagClient featureFlagClient;
+
+  public FeatureFlagFetchActivityImpl(final WorkspaceApi workspaceApi,
+                                      final FeatureFlagClient featureFlagClient) {
+    this.workspaceApi = workspaceApi;
+    this.featureFlagClient = featureFlagClient;
+  }
+
+  public UUID getWorkspaceId(final UUID connectionId) {
+    try {
+      final WorkspaceRead workspace = workspaceApi.getWorkspaceByConnectionId(new ConnectionIdRequestBody().connectionId(connectionId));
+      return workspace.getWorkspaceId();
+    } catch (final ApiException e) {
+      throw new RuntimeException("Unable to get workspace ID for connection", e);
+    }
+  }
+
+  @Override
+  public FeatureFlagFetchOutput getFeatureFlags(final FeatureFlagFetchInput input) {
+    final UUID workspaceId = getWorkspaceId(input.getConnectionId());
+
+    // TODO: remove this feature flag from here - not really needed by consumers but in here to get this
+    // activity up and running
+    final List<Flag> workspaceFlags = List.of(FieldSelectionEnabled.INSTANCE);
+    final Map<String, Boolean> featureFlags = new HashMap<>();
+    for (final Flag flag : workspaceFlags) {
+      featureFlags.put(flag.getKey(), featureFlagClient.enabled(flag, new Workspace(workspaceId)));
```

Comment:
Is it possible to do a bulk fetch? That would limit the number of calls we will do here. I don't know if there is any limitation with the lauch darkly API but this will be call as many time as we have a sync that runs.

Reply:
For later I believe that we should get those through an API, the same way we do for the DB. @colesnodgrass do you think that this would be too complex?

Reply:
My understanding of the feature flag client is that each evaluation of a feature flag does not represent an API call to LaunchDarkly since flags are cached by the SDK https://docs.launchdarkly.com/sdk/concepts/client-side-server-side#server-side-sdks-1

> Server-side SDKs receive the complete ruleset associated with an SDK key when they initialize a connection to LaunchDarkly's servers. LaunchDarkly continuously updates the SDK's cached flag ruleset whenever flag rules change on LaunchDarkly, using this persistent connection.
> 
> When your application requests flag evaluation, server-side SDKs evaluate feature flags using their cached ruleset, and determine the flag variation for a given evaluation context. To do this, they execute an in-process flag evaluation algorithm and return the resulting value.